pkgs/development/libraries/mesa/default.nix

```diff
@@ -65,6 +65,11 @@ stdenv.mkDerivation {
       url = "https://gitlab.freedesktop.org/mesa/mesa/commit/aebbf819df6d1e.patch";
       sha256 = "17248hyzg43d73c86p077m4lv1pkncaycr3l27hwv9k4ija9zl8q";
     })
+    # See: https://gitlab.freedesktop.org/mesa/mesa/-/issues/1020
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/XQuartz/XQuartz/master/src/mesa/mesa.patches/0004-Hack-to-address-build-failure-when-using-newer-macOS.patch";
```

Comment:
This should be e.g.: https://raw.githubusercontent.com/XQuartz/XQuartz/91f6b3d98ece6d566aa9f52b85aca1785f84f88e/src/mesa/mesa.patches/0004-Hack-to-address-build-failure-when-using-newer-macOS.patch

Links to the `master` branch aren't stable and the content could change at any time.

Reply:
Updated to use a file on disk to avoid this issue.

Reply:
Ok. AFAIK we prefer `fetchpatch` (https://nixos.org/manual/nixpkgs/unstable/#sec-patches, but IIRC not everyone agrees with that) and my link above should be stable but in this case adding the patch to Nixpkgs should be fine as the patch is small and we'd have to fetch it from a different repository (which the maintainers could theoretically delete or force-push to - shouldn't be a problem in this case though).